src/pages/main/tools/Editor/Editor.tsx

```diff
@@ -0,0 +1,68 @@
+import Draggable, { DraggableData, DraggableEvent } from 'react-draggable';
+import { LatexTextArea, PdfViewer } from '@components';
+
+import { RxDotsVertical } from 'react-icons/rx';
+import styles from './Editor.module.less';
+import { useResizeDetector } from 'react-resize-detector';
+import { useState } from 'react';
+
+const Editor = () => {
+  const DELIMITER_WIDTH = 5;
+  const MIN_PERCENTAGE_WIDTH = 0.3;
+
+  // Handle initial delimiter position
+  const { width, height, ref: rootRef } = useResizeDetector();
+  const [defaultDelimiterX, setDefaultDelimiterX] = useState(0.5);
+
+  const onStop = (_: DraggableEvent, data: DraggableData) => {
+    setDefaultDelimiterX(data.x / width);
+  };
+
+  interface DelimiterProps {
+    testId: string
+  }
+
+  const Delimiter = (props: DelimiterProps) => {
```

Comment:
are you planning on reusing this component somewhere else than 'Editor'? If yes, extract it to other component. If not, just do something like:

```
const delimiter = 
    <Draggable 
        axis='x'
        offsetParent={rootRef.current}
        defaultPosition={{ x: defaultDelimiterX * width - DELIMITER_WIDTH/2, y: 0 }}
        bounds={{
          left: MIN_PERCENTAGE_WIDTH * width,
          right: (1 - MIN_PERCENTAGE_WIDTH) * width - DELIMITER_WIDTH
        }}
        onStop={onStop}>
        <div
          className={styles.delimiter}
          style={{ height: height, width: DELIMITER_WIDTH }}
          data-testid='delimiter'>
          <RxDotsVertical
            size={16}
            className={styles.dragIcon}/>
        </div>
      </Draggable>
```

and then in return use `{ delimiter }` instead of `<Delimiter test-id={...} />`

Reply:
Not feasible, `Delimiter` is strongly connected with `Editor` component but making as you propose prevents it from updating when `width`, `height` and `defaultDelimiterX` change.

Reply:
then export `<Delimiter />` to `src/pages/main/editor/delimiter/Delimiter.tsx`